service/health/searcher/types.go

```diff
@@ -0,0 +1,9 @@
+package searcher
+
+type NodeInfo struct {
```

Comment:
Is there a reason for those types to be public?

Reply:
No, I have made the provider-independent cluster info structs private now.